### ISSUE TITLE
BUCK crashes on the latest NDK

### ISSUE THREAD

**Issue by hzsweers** [facebook/buck#1890]
with NDK 17. Some architectures were removed

```
UNKNOWN ERROR: com.google.common.util.concurrent.ExecutionError: java.lang.AssertionError: /Users/zsweers/dev/android/android-sdk/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi doesn't exist.
com.google.common.util.concurrent.ExecutionError: com.google.common.util.concurrent.ExecutionError: java.lang.AssertionError: /Users/zsweers/dev/android/android-sdk/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi doesn't exist.
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2216)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4147)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4151)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5140)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.getOrCreate(DefaultToolchainProvider.java:166)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.getByName(DefaultToolchainProvider.java:109)
	at com.facebook.buck.toolchain.BaseToolchainProvider.getByName(BaseToolchainProvider.java:24)
	at com.facebook.buck.jvm.java.JavaBinaryDescription.getCxxPlatform(JavaBinaryDescription.java:79)
	at com.facebook.buck.jvm.java.JavaBinaryDescription.findDepsForTargetFromConstructorArgs(JavaBinaryDescription.java:175)
	at com.facebook.buck.jvm.java.JavaBinaryDescription.findDepsForTargetFromConstructorArgs(JavaBinaryDescription.java:55)
	at com.facebook.buck.rules.TargetNodeFactory.create(TargetNodeFactory.java:115)
	at com.facebook.buck.rules.TargetNodeFactory.createFromObject(TargetNodeFactory.java:65)
	at com.facebook.buck.parser.DefaultParserTargetNodeFactory.createTargetNodeFromObject(DefaultParserTargetNodeFactory.java:233)
	at com.facebook.buck.parser.DefaultParserTargetNodeFactory.createTargetNode(DefaultParserTargetNodeFactory.java:152)
	at com.facebook.buck.parser.DefaultParserTargetNodeFactory.createTargetNode(DefaultParserTargetNodeFactory.java:61)
	at com.facebook.buck.parser.TargetNodeParsePipeline.computeNode(TargetNodeParsePipeline.java:124)
	at com.facebook.buck.parser.TargetNodeParsePipeline.computeNode(TargetNodeParsePipeline.java:54)
	at com.facebook.buck.parser.ConvertingPipeline.dispatchComputeNode(ConvertingPipeline.java:166)
	at com.facebook.buck.parser.ConvertingPipeline.lambda$null$0(ConvertingPipeline.java:83)
	at com.facebook.buck.parser.PipelineNodeCache.getJobWithCacheLookup(PipelineNodeCache.java:73)
	at com.facebook.buck.parser.ConvertingPipeline.lambda$getAllNodesJob$1(ConvertingPipeline.java:76)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:206)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:195)
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:115)
	at com.google.common.util.concurrent.MoreExecutors$5$1.run(MoreExecutors.java:999)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.google.common.util.concurrent.ExecutionError: java.lang.AssertionError: /Users/zsweers/dev/android/android-sdk/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi doesn't exist.
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2216)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4147)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4151)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5140)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.getOrCreate(DefaultToolchainProvider.java:166)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.isToolchainPresent(DefaultToolchainProvider.java:125)
	at com.facebook.buck.cxx.toolchain.CxxPlatformsProviderFactory.createToolchain(CxxPlatformsProviderFactory.java:47)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.createToolchain(DefaultToolchainProvider.java:191)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.access$100(DefaultToolchainProvider.java:46)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider$1.load(DefaultToolchainProvider.java:64)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider$1.load(DefaultToolchainProvider.java:58)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3708)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2416)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2299)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2212)
	... 27 more
Caused by: java.lang.AssertionError: /Users/zsweers/dev/android/android-sdk/ndk-bundle/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi doesn't exist.
	at com.facebook.infer.annotation.Assertions.assertCondition(Assertions.java:72)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms$NdkCxxToolchainPaths.processPathPattern(NdkCxxPlatforms.java:963)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms$NdkCxxToolchainPaths.getCxxRuntimeLibsDirectory(NdkCxxPlatforms.java:1053)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms$NdkCxxToolchainPaths.access$000(NdkCxxPlatforms.java:877)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms.getCcLinkTool(NdkCxxPlatforms.java:696)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms.build(NdkCxxPlatforms.java:477)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms.getPlatforms(NdkCxxPlatforms.java:274)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms.getPlatforms(NdkCxxPlatforms.java:239)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatforms.getPlatforms(NdkCxxPlatforms.java:216)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatformsProviderFactory.getNdkCxxPlatforms(NdkCxxPlatformsProviderFactory.java:63)
	at com.facebook.buck.android.toolchain.ndk.impl.NdkCxxPlatformsProviderFactory.createToolchain(NdkCxxPlatformsProviderFactory.java:41)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.createToolchain(DefaultToolchainProvider.java:191)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider.access$100(DefaultToolchainProvider.java:46)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider$1.load(DefaultToolchainProvider.java:64)
	at com.facebook.buck.toolchain.impl.DefaultToolchainProvider$1.load(DefaultToolchainProvider.java:58)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3708)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2416)Connection is lost to Buck daemon! This usually indicates that daemon experienced an unrecoverable error. Here is what you can do:
```

**Comment by hzsweers** [comment]
CC @kageiit

**Comment by kageiit** [comment]
https://github.com/facebook/buck/pull/1889#discussion_r190300732 should fix the issue

**Comment by styurin** [comment]
Fixed in 4ef8e96bae72988f4bd77994a0228a595aa42ac6.
